validator/client/propose_protect.go

```diff
@@ -33,17 +35,19 @@ func (v *validator) preBlockSignValidations(ctx context.Context, pubKey [48]byte
 		)
 	}
 
-	_, exists, err = v.db.ProposalHistoryForSlot(ctx, pubKey, block.Slot)
+	prevSigningRoot, exists, err := v.db.ProposalHistoryForSlot(ctx, pubKey, block.Slot)
 	if err != nil {
 		if v.emitAccountMetrics {
 			ValidatorProposeFailVec.WithLabelValues(fmtKey).Inc()
 		}
 		return errors.Wrap(err, "failed to get proposal history")
 	}
-	// If a proposal exists in our history for the slot, we assume it is slashable.
-	// TODO(#7848): Add a more sophisticated strategy where if we indeed have the signing root,
-	// only blocks that have a conflicting signing root with a historical proposal are slashable.
-	if exists {
+
+	// If a proposal exists in our history for the slot, we check the following:
+	// If the signing root is nil, then we consider it slashable. If signing root is not nil,
```

Comment:
There is no check if the signing root is nil below. Did you mean if it's zero hash? What if it is `nil`?

Reply:
Cannot be nil because the return type is [32]byte